### PR TITLE
Adding docker support for basic example

### LINKED ISSUE #16: Dockerfile?
Is there a Dockerfile for building owt-server? I'm getting a few compile errors and using Docker as a standardized way to build a binary would be nice.

I started writing one.. and happy to share it.. But its getting stuck running the scripts/build.js building the webrtc code.. I think it is trying to build licode, but its missing the r_types.h header file.

Which ubuntu is best? 

Is there a demo server somewhere? 

FYI, I'm mostly interested in IP Camera RTSP broadcasting to 0 to N viewers. Saw your server supports RTSP, but not sure if it supports rtsp feeds as a webrtc source. 

Thanks!

### PR DESCRIPTION
**Pull request by jimmy-huang**
Adding Dockerfile and use docker compose to build owt-server in the docker container.  You can simply run docker-compose build to build the base image and docker-compose up to start the server with the conference sample, this helps resolve #16.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+RUN git clone https://github.com/open-webrtc-toolkit/owt-server.git
+WORKDIR /owt-server
+
+# This is needed to patch licode
+RUN  git config --global user.email "you@example.com" && \
+  git config --global user.name "Your Name"
+
+RUN ./scripts/installDepsUnattended.sh
+RUN ./scripts/build.js -t all --check
+RUN ./scripts/pack.js -t all --install-module --sample-path $CLIENT_SAMPLE_PATH .
```
Comment (qwu16) [inline docker/Dockerfile]: Please remove build.js and pack.js in the Dockerfile. These 2 scripts will be frequently executed for future MCU developers, so it's better to wrap them in a script and then developers can execute the script to finish build and pack process in the container. We should provide a clean MCU build environment as base docker image.

Comment (jimmy-huang) [reply docker/Dockerfile]: @qwu16, maybe those 2 lines should be in another Dockerfile for the app itself, and this Dockerfile will be just the server.  Then docker-compose would pick the app container instead.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+RUN cd dist/extras/basic_example && npm install
```
[the remark this answers is not in the document]
Comment (jimmy-huang) [reply docker/Dockerfile]: It seems that the install_openh264.sh script doesn't install properly, might need to fix that first, see #104.

File: docker/Dockerfile (hunk 1 above)
Comment (jimmy-huang) [reply docker/Dockerfile]: I moved the build.js and pack.js steps into a separate Dockerfile, the main Dockerfile just builds the server, and the docker-compose.yml would build both of them.

Comment (qwu16) [reply docker/Dockerfile]: LGTM

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+RUN git clone https://github.com/open-webrtc-toolkit/owt-server.git
+WORKDIR /owt-server
+
+# This is needed to patch licode
+RUN  git config --global user.email "you@example.com" && \
+  git config --global user.name "Your Name"
+
+RUN ./scripts/installDepsUnattended.sh
+RUN ./source/agent/video/install_openh264.sh
```
Comment (qwu16) [inline docker/Dockerfile]: this line should move to the end of Dockerfile-conference, and changed to Run ./dist/video_agent/install_openh264.sh, for we need to install openh264 when running MCU, related openh264 dependency for packing MCU has been installed in scripts/installDepsUnattended.sh

Comment (jimmy-huang) [reply docker/Dockerfile]: @qwu16 thanks for feedback, I updated the patch

**Pull request merged**